python/lsst/daf/butler/_butler.py

```diff
@@ -2415,13 +2415,22 @@ def transfer_from(
 
             # Ask the datastore to transfer. The datastore has to check that
             # the source datastore is compatible with the target datastore.
-            self.datastore.transfer_from(
+            accepted, rejected = self.datastore.transfer_from(
                 source_butler.datastore,
                 source_refs,
                 local_refs=transferred_refs,
                 transfer=transfer,
                 artifact_existence=artifact_existence,
             )
+            if rejected:
+                # For now, accept the registry entries but not the files.
```

Comment:
Not sure what this comment means 🤔

Reply:
It means I'm not now going to try to remove the entries from the destination registry for all the times that datastore couldn't receive the actual file. We could go back and remove the entries from Registry (maybe we should). @TallJimbo do you have an opinion on what happens if you transfer from A to B but datastore in B rejects doesn't accept the files because of constraints? We can (a) raise an exception because you asked us to transfer something and it didn't work (b) have registry entries but no datastore records (c) remove the registry entries as well. Also, should we raise if this means that nothing was transferred?

Reply:
Transferring registry content and dropping the datastore content seems like the best course of action, but I'm also not opposed to raising (and we might be able to do that up front, just based on the dataset types being transferred and the constraints).

Reply:
Ok. Will leave it as is.